src/ShaderAsset.cpp

```diff
@@ -0,0 +1,88 @@
+#include "ShaderAsset.h"
+
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+bool ShaderAsset::load_shader(const std::string& vertex_path, const std::string& fragment_path) {
+    // Create the shaders
+    GLuint vertex_shader_id = glCreateShader(GL_VERTEX_SHADER);
+    GLuint fragment_shader_id = glCreateShader(GL_FRAGMENT_SHADER);
+
+    std::string VertexShaderCode;
```

Comment:
This variable should be `snake_case` and I think it would be better to explicitly initialize it.